sdks/java/core/src/test/java/com/google/cloud/dataflow/sdk/io/AvroIOTest.java

```diff
@@ -222,6 +222,37 @@ public void testAvroSinkWrite() throws Exception {
     }
   }
 
-  // TODO: for Write only, test withSuffix, withNumShards,
+  @SuppressWarnings("deprecation") // using AvroCoder#createDatumReader for tests.
+  @Test
+  public void testAvroSinkShardedWrite() throws Exception {
+    String outputFilePrefix = new File(tmpFolder.getRoot(), "prefix").getAbsolutePath();
+    String[] expectedElements = new String[] {"first", "second", "third", "fourth", "fifth"};
+
+    TestPipeline p = TestPipeline.create();
+    int numShards = 4;
+    p.apply(Create.<String>of(expectedElements))
+        .apply(
+            AvroIO.Write.to(outputFilePrefix)
+                .withSchema(String.class)
+                .withNumShards(numShards)
+                .withShardNameTemplate(ShardNameTemplate.INDEX_OF_MAX));
+    p.run();
+
+    // Validate that the data written matches the expected elements in the expected order
+    List<String> actualElements = new ArrayList<>();
```

Comment:
Some code duplication between tests. Perhaps refactor to use the same code path.